src/main/java/nguyen/Storage.java

```diff
@@ -0,0 +1,67 @@
+package nguyen;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Handles loading and saving tasks to a file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file and returns them as an ArrayList.
+     *
+     * @return An ArrayList of Task objects loaded from the file.
+     * @throws NguyenException If an error occurs while reading the file.
+     */
+    public ArrayList<Task> load() throws NguyenException {
+        ArrayList<Task> taskList = new ArrayList<>();
+        String line;
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            while ((line = reader.readLine()) != null) {
+                Task task = Parser.parseTask(line);
+                if (task != null) {
+                    taskList.add(task);
+                }
+            }
+        } catch (IOException e) {
+            throw new NguyenException(e.getMessage());
+        }
+        return taskList;
+    }
+
+    /**
+     * Saves the tasks from the TaskList to the file.
+     *
+     * @param taskList The TaskList containing tasks to be saved.
+     */
+    public void saveTask(TaskList taskList) {
+        try {
+            // Clear the existing file content
+            new FileWriter(filePath, false).close();
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+                for (int i = 0; i < taskList.size(); i++) {
+                    writer.write(taskList.get(i).toString());
+                    writer.newLine();
+                }
+            }
```

Comment:
I think the first try block is redundant and can be removed. You can instead open the BufferedWriter with false for append mode, which should already overwrite the file
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath, false))) {
            for (int i = 0; i < taskList.size(); i++) {
                writer.write(taskList.get(i).toString());
                writer.newLine();
            }
        }